### PR TITLE
[Reviewer: Ellie] Print out Ellis errors in more detail

### PR DESCRIPTION
**Pull request by rkd-msw**
This just makes it easier to diagnose issues like https://github.com/Metaswitch/ellis/issues/174 and https://github.com/Metaswitch/ellis/issues/175, by catching the exceptions and printing out more HTTP-level detail.

Output looks like:
```
$ bundle exec rake test[cw-ngv.com] TESTS="Basic Call - Mainline"
Basic Call - Mainline (TCP) - (6505550024, 6505550050) Passed
Basic Call - Mainline (UDP) - (6505550051, 6505550097) Passed
Listing existing numbers failed with HTTP code 502, body {"status": 502, "message": "Bad Gateway", "reason": "Upstream request failed", "detail": {"Upstream error": "HTTP 404: Not Found"}, "error": true}
```

### CODE REVIEW

File: lib/ellis.rb [hunk 1]
```diff
@@ -54,9 +54,14 @@ def self.destroy_leaked_numbers(domain)
...
     cookie = r.cookies
-    r = RestClient::Request.execute(method: :get,
-                                    url: ellis_url(domain, "accounts/#{EMAIL}/numbers"),
-                                    cookies: cookie)
+    begin
+      r = RestClient::Request.execute(method: :get,
+                                      url: ellis_url(domain, "accounts/#{EMAIL}/numbers"),
+                                      cookies: cookie)
+    rescue RestClient::Exception => e
+      puts "Listing existing numbers failed with HTTP code #{e.http_code}, body #{e.http_body}"
```
Comment (eleanor-merry) [inline lib/ellis.rb]: +1 for stopping this crashing.

I'm not convinced that the body is useful information though (it's hard to parse and you probably need to go to the Ellis logs anyway to get the context)

**Comment by eleanor-merry** [comment]
Looks good, but I think that printing the body as well as the error code actually makes it harder to understand

File: lib/ellis.rb (hunk 1 above)
Comment (rkd-msw) [inline lib/ellis.rb]: I think the detail of the body is useful, especially after my other PR adding the upstream URL - being able to see at a glance that this live test error is because a request to Homestead's /associated_private_ids URL returned 404 feels like a good thing. I will limit it to just the detail section, and put it on a newline, though.
